aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1377,7 +1413,27 @@ AT_ERROR("svd: MAGMA library not found in "
 
   magma_int_t m = magma_int_cast(self.size(-2), "m");
   magma_int_t n = magma_int_cast(self.size(-1), "n");
-  auto k = std::min(m, n);
+  auto mn = std::min(m, n);
+
+  c10::Storage storage_rwork;
+  value_t* rwork = nullptr;
+
+  magma_int_t* iwork;
+  ALLOCATE_ARRAY(iwork, magma_int_t, 8 * mn);
+  // Copy-n-paste rwork size computation from BatchLinearAlgebra.cpp
+  if (isComplexType(at::typeMetaToScalarType(self.dtype()))) {
+    auto mx = std::max(m, n);
+    int64_t lrwork; // These settings are valid for on LAPACK 3.6+
+    if (jobz == MagmaNoVec){
+      lrwork = 7 * mn;
+    } else if (mx > 10 * mn){
```

Comment:
`mx > 10 * n` was an arbitrary decision, since LAPACK seems to not document this number https://github.com/pytorch/pytorch/pull/27653#discussion_r336700894.
But [MAGMA documents](https://icl.cs.utk.edu/projectsfiles/magma/doxygen/group__magma__gesdd.html) this condition. The threshold for mx >> mn is currently `mx >= mn*17/9`.
[In code, this condition is calculated with](https://bitbucket.org/icl/magma/src/f9aba3b1a6926a6927b2d5aa0bb7d1db6b144da5/testing/testing_zgesdd.cpp#lines-528) `(magma_int_t) (mn * 17. / 9.)`, maybe we can make it `mx >= mn * 2`?

Reply:
Do you know how `17/9` ration is computed? As in the documentation is simply says `if mx >> mn`

Reply:
I don't know.